pkg/operator/ceph/csi/spec.go

```diff
@@ -73,6 +73,8 @@ type Param struct {
 	CSICephFSPodLabels             map[string]string
 	CSINFSPodLabels                map[string]string
 	CSIRBDPodLabels                map[string]string
+	// TODO: do we want labels to be customizable?
```

Comment:
@BlaineEXE @travisn thoughts?

Reply:
To summarize the main two options we discussed:
1. Two holder pods (one for rbd and one for cephfs). The tolerations and affinity would just be inherited from the corresponding rbd or cephfs plugins.
2. Keep a single holder pod per node. The tolerations and affinity would be defined as their own settings.

Option 2 is compelling to me to keep the number of pods down, but option 1 is compelling to keep the settings simpler, which is less burden on the admin to get the settings just right. I'm not sure I have a strong preference. Are there any other factors to consider that would make one choice more obvious?

Reply:
Thanks for the summary, there is a hiccup with the 2 Holder approach, today the csi implementation only expects a single network namespace per cluster. If we want to go that route we would need to decouple even further. Having a single DS is less flexible and a bit more prone to errors if one of the drivers (RBD or CephFS) gets enabled/disabled I guess.
Having 2 holders makes the configuration a bit more straightforward as users don't need to worry about placement/tolerations, they will be inherited from the enabled drivers. I'm leaning towards option 1 at the moment. I need to discuss with @Madhu-1 about the CSI change too.

Reply:
Yes, Agree having option 1 keeps it single. https://github.com/ceph/ceph-csi/pull/3039 should provide an option to specify cephfs and rbd net file path.